### DISCUSSION TITLE
Wayland failing to start

### DISCUSSION THREAD

**Discussion by smsteward** [qtile/qtile#3301] [Q&A]
I figured I'd make this a discussion and not an issue, since I believe this is on my end and not Qtile's.

I am unable to start Qtile with the Wayland backend by any means. I use LightDM as my login manager, and selecting the "Qtile (Wayland) option from the sessions menu and logging in results in a black screen for about 10-15 seconds and being taken back to the LightDM login screen.

If I try to start the Wayland session in a nested window from within X with `qtile start -b wayland`, I get `[backend/x11/backend.c:609] Failed to query DRI3 DRM FD` at the top before the Traceback, and the Traceback itself states `RuntimeError: Failed to create wlroots backend`

Neither of these failures write any information to the Qtile log file, so I can't diagnose it that way. Also, FWIW, I'm using an Nvidia graphics card.

Here are my package and OS stats, showing that I do currently have all the Wayland dependencies for both Python and the system installed and at latest versions.

I'm not sure if I need to manually compile a backend; my initial assumption was that simply having the packages installed would suffice. Any help would be most appreciated, and thanks to the contributors and maintainers for such an amazing window manager!

OS: Arch Linux
Graphics: Nvidia 2070 SUPER on driver version 510.47.03
egl-wayland: 2:1.1.9+r3+g582b2d3-1
python-pywayland: 0.4.9-1
wayland: 1.20.0-1
wayland-protocols: 1.25.1
xorg-xwayland 21.1.4-1
python-pywlroots: 0.15.7-2
wlroots: 0.15.0-4
python-xkbcommon: 0.4-3

**Comment by m-col** [comment]
Sorry to hear you are having problems with it. You're right that you should just need the packages, no compiling needed on your end. All your dependency versions look fine. The nvidia driver is absolutely suspect, as this has been a headache for wlroots in the past, though I see that some headway has been made (on nvidia's side) to make this situation less bad. You could likely rule this out though, if you can successfully run other wlroots-based compositors just fine. Have you tested/can you test running sway?
